Build failure on Fedora 44: variable set but not used

### Is there an already existing issue for this?

- [x] I have searched the existing issues

### Expected behavior

Build passes, no warnings are generated.

### Current behavior

Build fails with:

```
[  108s] /home/abuild/rpmbuild/BUILD/fast-dds-3.6.1-build/Fast-DDS-3.6.1/include/fastdds/rtps/common/CacheChange.hpp: In member function ‘void eprosima::fastdds::rtps::CacheChange_t::setFragmentSize(uint16_t, bool)’:
[  108s] /home/abuild/rpmbuild/BUILD/fast-dds-3.6.1-build/Fast-DDS-3.6.1/include/fastdds/rtps/common/CacheChange.hpp:270:24: error: variable ‘offset’ set but not used [-Werror=unused-but-set-variable=]
[  108s]   270 |                 size_t offset = 0;
[  108s]       |                        ^~~~~~
```

[Fast DDS source](https://github.com/eProsima/Fast-DDS/blob/43d6be76bb15b781ecd988f2c6f8a905f67cc3ff/include/fastdds/rtps/common/CacheChange.hpp#L270-L271)

Suggested fix is to remove `offset`, which is incremented but indeed never read.

### Steps to reproduce

I try to build myself an RPM on Fedora 44 using [OBS](https://openbuildservice.org/). `-Werror=unused-but-set-variable=` is set.

### Fast DDS version/commit

v3.6.1

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Built on Fedora 44

### XML configuration file

```xml

```

### Relevant log output

```shell

```

### Network traffic capture

_No response_

--- a/include/fastdds/rtps/common/CacheChange.hpp
+++ b/include/fastdds/rtps/common/CacheChange.hpp
@@ -267,8 +267,7 @@ struct FASTDDS_EXPORTED_API CacheChange_t
             {
                 // Keep index of next fragment on the payload portion at the beginning of each fragment. Last
                 // fragment will have fragment_count_ as 'next fragment index'
-                size_t offset = 0;
-                for (uint32_t i = 1; i <= fragment_count_; i++, offset += fragment_size_)
+                for (uint32_t i = 1; i <= fragment_count_; i++)
                 {
                     set_next_missing_fragment(i - 1, i);  // index to next fragment in missing list
                 }

